cpp/dolfinx/CMakeLists.txt

```diff
@@ -147,7 +147,7 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/common/version.h DESTINATION ${CMAKE_I
 include(CMakePackageConfigHelpers)
 write_basic_package_version_file(${CMAKE_BINARY_DIR}/dolfinx/DOLFINXConfigVersion.cmake
   VERSION ${DOLFINX_VERSION}
-  COMPATIBILITY ExactVersion)
+  COMPATIBILITY AnyNewerVersion)
```

Comment:
Why not same minor version?

Reply:
I agree with Jack, `COMPATIBILITY SameMinorVersion` is I think more appropriate, at least at this stage in dolfinx' development. Otherwise 0.6.0 would be considered compatible with 0.5.2.

Reply:
That does not work with the version ranges unfortunately.

Reply:
Does SameMinorVersion give the same problem as ExactVersion? In that case fair enough.  If AnyNewerVersion is used here, then it would be possible to exclude 0.6 in the client program by using `...<0.6` with the find_package version.

Reply:
I will test it again tomorrow, but as far as I can recall it had the same issues. I would say it is up to the developer of the other package (me in the case of DOLFINx-MPC) to set sensible ranges in `find_package` in my cmake file, similarly to how you would pin, or set version ranges in Python packages.

Reply:
Likely you already thought of it, but one thought in regards to testing `SameMinorVersion`, both packages (dolfinx and dolfinx_mpc) would need `cmake_minimum_required(VERSION 3.19)`.  i.e. dolfinx_mpc would also need that updated alongside the version range.